base/show.jl

```diff
@@ -992,13 +992,34 @@ function show(io::IO, l::Core.MethodInstance)
             show(io, def)
         else
             print(io, "MethodInstance for ")
-            show_tuple_as_call(io, def.name, l.specTypes)
+            show_tuple_as_call(io, def.name, l.specTypes, false, nothing, nothing, true)
         end
     else
         print(io, "Toplevel MethodInstance thunk")
     end
 end
 
+function show(io::IO, mi_info::Core.Compiler.Timings.InferenceFrameInfo)
+    mi = mi_info.mi
+    def = mi.def
+    if isa(def, Method)
+        if isdefined(def, :generator) && mi === def.generator
+            print(io, "InferenceFrameInfo generator for ")
+            show(io, def)
+        else
+            print(io, "InferenceFrameInfo for ")
+            argnames = [isa(a, Core.Const) ? a.val : "" for a in mi_info.slottypes]
+            show_tuple_as_call(io, def.name, mi.specTypes, false, nothing, argnames, true)
+        end
+    else
+        print(io, "Toplevel InferenceFrameInfo thunk")
```

Comment:
Is there nothing else that can be usefully printed, here?

Reply:
I figured something out. Now it prints `thunk from $module starting at $file:$line` which seems like an improvement.

Reply:
Amazing. Me too. Thanks!